ci: drop gitleaks job (organization license unavailable)

## Context

The `secret-scan` job (named `gitleaks`) in `.github/workflows/ci.yml` consistently fails on PRs with:

```
[Glaon-Smart] is an organization. License key is required.
##[error]🛑 missing gitleaks license. Go grab one at gitleaks.io and store it as a GitHub Secret named GITLEAKS_LICENSE.
```

gitleaks-action v2 requires a paid org license. Without it the action exits non-zero, blocking every PR via the required-status-check rule.

## Decision

Drop the gitleaks job from CI and from both branch protection rulesets. Secret scanning can be re-introduced later behind a different tool (e.g. trufflehog) or once a gitleaks license is procured.

## Scope

- Remove `secret-scan` job from `.github/workflows/ci.yml`
- Remove `{ "context": "gitleaks" }` entry from `.github/rulesets/development.json`
- Remove `{ "context": "gitleaks" }` entry from `.github/rulesets/main.json`
- Run `scripts/apply-rulesets.sh` (admin token) so the new ruleset takes effect on GitHub

## Out of scope

- Choosing a replacement secret-scanning tool — separate issue when needed.

--- a/.github/rulesets/development.json
+++ b/.github/rulesets/development.json
@@ -37,9 +37,6 @@
             "context": "conventional-commits"
           },
           {
-            "context": "gitleaks"
-          },
-          {
             "context": "i18n-check"
           },
           {

--- a/.github/rulesets/main.json
+++ b/.github/rulesets/main.json
@@ -34,9 +34,6 @@
         "strict_required_status_checks_policy": true,
         "required_status_checks": [
           {
-            "context": "gitleaks"
-          },
-          {
             "context": "i18n-check"
           },
           {

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -132,17 +132,6 @@ jobs:
       - name: Build web + enforce bundle-size budget
         run: pnpm size
 
-  secret-scan:
-    name: gitleaks
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@34e114876b0b11c390a56381ad16ebd13914f8d5 # v4.3.1
-        with:
-          fetch-depth: 0
-      - uses: gitleaks/gitleaks-action@ff98106e4c7b2bc287b24eaf42907196329070c7 # v2.3.9
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
   commit-lint:
     name: conventional-commits
     # Skipped for release PRs (development → main): those bundle historical

--- a/apps/web/src/features/setup/setup-route.tsx
+++ b/apps/web/src/features/setup/setup-route.tsx
@@ -73,7 +73,7 @@ const STEP_ICON_PATHS: Record<WizardStepId, string> = {
 function StepIcon({ id }: { id: WizardStepId }): ReactNode {
   return (
     <svg
-      className="size-5 text-secondary"
+      className="size-5"
       viewBox="0 0 24 24"
       fill="none"
       stroke="currentColor"

--- a/docs/dependencies.md
+++ b/docs/dependencies.md
@@ -217,6 +217,17 @@ Bir paket yeni sürüm sonrası sorun çıkarırsa:
    ```
 3. Ayrı issue aç, kök sebebi araştır, hazır olunca constraint'i kaldır.
 
+Transitive bir bağımlılık high/critical advisory aldıysa ve doğrudan tüketicimiz henüz patch sürümü yayınlamadıysa, aynı yöntemi `pnpm.overrides` üzerinden kullanırız — kök `package.json`'da `pnpm.overrides` bloğuna minimum güvenli versiyonu yaz, `pnpm install` ile lockfile'ı yenile, audit'in yeşil olduğunu doğrula. Her aktif override aşağıdaki tabloda gerekçesi ve kalkış tetikleyicisi ile birlikte tutulur — pin gereksizleştiğinde (override olmadan da `pnpm audit --audit-level high` yeşil) `package.json` ve bu tablodan birlikte düşürülür.
+
+### Aktif override'lar
+
+Kök [`package.json`](../package.json) `pnpm.overrides` bloğunda hâlen aktif olan pin'ler:
+
+| Paket       | Override  | Gerekçe                                                                                                                                           | Ne zaman düşer                                                                            | Tracking |
+| ----------- | --------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- | -------- |
+| `basic-ftp` | `>=5.3.1` | [GHSA-rpmf-866q-6p89](https://github.com/advisories/GHSA-rpmf-866q-6p89) — transitive `basic-ftp@<5.3.1`                                          | `basic-ftp` transitive consumer'ları kendiliğinden `>=5.3.1`'e geçtiğinde                 | #405     |
+| `js-cookie` | `>=3.0.7` | [GHSA-qjx8-664m-686j](https://github.com/advisories/GHSA-qjx8-664m-686j) — `apps/mobile` Clerk SDK üzerinden transitive `js-cookie@3.0.5` çekiyor | Clerk SDK kendi içinde `js-cookie >=3.0.7`'ye geçtiğinde (`pnpm why js-cookie` ile teyit) | #580     |
+
 ## Sorun giderme
 
 - **Dashboard issue yok** → Renovate GitHub App repo'ya yüklü mü? Repo → Settings → Integrations.

--- a/package.json
+++ b/package.json
@@ -34,7 +34,8 @@
   "packageManager": "pnpm@9.15.9",
   "pnpm": {
     "overrides": {
-      "basic-ftp": ">=5.3.1"
+      "basic-ftp": ">=5.3.1",
+      "js-cookie": ">=3.0.7"
     }
   },
   "private": true,

--- a/packages/ui/src/components/SetupStepNav/SetupStepNav.tsx
+++ b/packages/ui/src/components/SetupStepNav/SetupStepNav.tsx
@@ -136,13 +136,16 @@ function SetupStepNavItem({ step, state, isLast, onSelect }: SetupStepNavItemPro
   const titleClass = `text-sm font-semibold leading-5 ${isActive ? 'text-brand-secondary' : 'text-tertiary'}`;
   const descriptionClass = 'text-sm font-normal leading-5 text-tertiary';
   const iconNode = isCompleted ? <Check aria-hidden="true" /> : step.icon;
-  // Icon container: active gets a filled brand-coloured tile with a
-  // white glyph (loud emphasis); completed + upcoming stay on the
-  // neutral white surface so the active row is the only visually
-  // "loud" element in the rail.
-  const iconContainerClass = isActive
-    ? 'z-[1] flex size-10 shrink-0 items-center justify-center rounded-lg bg-brand-solid text-white shadow-xs-skeuomorphic ring-1 ring-brand ring-inset'
-    : 'z-[1] flex size-10 shrink-0 items-center justify-center rounded-lg bg-primary text-secondary shadow-xs-skeuomorphic ring-1 ring-primary ring-inset';
+  const base =
+    'z-[1] flex size-10 shrink-0 items-center justify-center rounded-lg shadow-xs-skeuomorphic ring-1 ring-inset';
+  let iconContainerClass: string;
+  if (isActive) {
+    iconContainerClass = `${base} bg-brand-solid text-white ring-brand`;
+  } else if (isCompleted) {
+    iconContainerClass = `${base} bg-success-solid text-white ring-[var(--color-bg-success-solid)]`;
+  } else {
+    iconContainerClass = `${base} bg-primary text-secondary ring-primary`;
+  }
   const textBlock = (
     <div
       className="flex flex-1 flex-col pb-8 text-left"

